web/src/emoji.ts

```diff
@@ -1,34 +1,93 @@
 import _ from "lodash";
 
+type RealmEmoji = Record<
+    string,
+    {
+        id: number;
+        author_id: number;
+        deactivated: boolean;
+        name: string;
+        source_url: string;
+        still_url: string | null;
+    }
+>;
+
+type EmojiCodes = {
+    codepoint_to_name: Record<string, string>;
+    name_to_codepoint: Record<string, string>;
+    emoji_catalog: Record<string, string[]>;
+    emoticon_conversions: Record<string, string>;
+    names: string[];
+};
+
+type EmojiParams = {
+    realm_emoji: RealmEmoji;
+    emoji_codes: EmojiCodes;
+};
+
+type Translation = {
+    regex: RegExp;
+    replacement_text: string;
+};
+
+type Emoji = {
+    id: number;
+    emoji_name: string;
+    emoji_url: string;
+    still_url: string | null;
+    deactivated: boolean;
+};
+
+type EmojiDict = {
+    name: string;
+    display_name: string;
+    aliases: string[];
+    is_realm_emoji: boolean;
+    has_reacted: boolean;
+    emoji_code?: string;
+    url?: string;
+};
+
+type EmojiInfo = {
```

Comment:
Let's call this `EmojiRenderingDetails`, since the functions that return one seem to think they're returning "emoji rendering details". Probably in addition to giving these better names, we should also add comments for each of these types describing what they mean logically.